syncthing.if

```diff
@@ -0,0 +1,55 @@
+########################################
+##	<summary>
+##	Role access for Syncthing
+##	</summary>
+##	<param name="role">
+##		<summary>
+##			Role allowed access
+##		</summary>
+##	</param>
+##	<param name="domain">
+##		<summary>
+##			User domain for the role
+##		</summary>
+##	</param>
+#
+interface(`syncthing_role', `
+
+	gen_require(`
+		attribute_role syncthing_roles;
+		type syncthing_t, syncthing_exec_t, syncthing_config_home_t;
+	')
+
+	# Declarations
+	roleattribute $1 syncthing_roles;
+
+	# Policy
+	domtrans_pattern($2, syncthing_exec_t, syncthing_t)
+')
```

Comment:
On 08/21/2016 11:03 PM, Naftuli Tzvi Kay wrote:
>> +##		</summary>
>> +##	</param>
>> +#
>> +interface(`syncthing_role', `
>> +
>> +	gen_require(`
>> +		attribute_role syncthing_roles;
>> +		type syncthing_t, syncthing_exec_t, syncthing_config_home_t;
>> +	')
>> +
>> +	# Declarations
>> +	roleattribute $1 syncthing_roles;
>> +
>> +	# Policy
>> +	domtrans_pattern($2, syncthing_exec_t, syncthing_t)
>> +')
> 
> @doverride so should I do this:
> 
> ```
> role $1 types syncthing_config_home_t;
> userdom_user_home_dir_filetrans($2, syncthing_config_home_t, dir, "syncthing")
> ```
> 
> Since `syncthing_config_home_t` has the `user_home_content` attribute, it should be able to be r/w by the user.
> 

No, that is a property specific to the RedHat refpolicy fork.

Something like:

allow $2 syncthing_config_home_t:dir { manage_dir_perms relabel_dir_perms };
allow $2 syncthing_config_home_t:file { manage_file_perms
relabel_file_perms };

-- 
Key fingerprint = 5F4D 3CDB D3F8 3652 FBD8  02D5 3B6C 5F1D 2C7B 6B02
https://sks-keyservers.net/pks/lookup?op=get&search=0x3B6C5F1D2C7B6B02
Dominick Grift